lib/resty/kafka/ringbuffer.lua

```diff
@@ -46,6 +75,19 @@ function _M.add(self, topic, key, message)
 end
 
 
+function _M.release_buffer_wait(self)
+    if not self.wait_on_buffer_full then
+        return nil
+    end
+
+    if self.sema:count() < 0 then
+        self.sema:post(1)
```

Comment:
please add a comment here, why post(1) is enough, when there are multiple waiters, for better understanding.